release/release-offline.ts

```diff
@@ -0,0 +1,410 @@
+      error(`You must build the jar first. ${file} does not exist`);
+    }
+  });
+
+  const versionProperties = (
+    await $`jar xf ${JAR_PATH}/metabase.jar version.properties && cat version.properties`
+  ).toString();
+
+  log(versionProperties, "green");
+
+  if (!versionProperties.includes(`tag=${version}`)) {
+    error(`This jar does not match the input version: ${version}`);
+  }
+}
+
+// eslint-disable-next-line no-console
+const log = (message, color = "blue") =>
+  console.log(chalk[color](`\n${message}\n`));
+
+function error(message) {
+  log(`⚠️   ${message}`, "red");
+  process.exit(1);
+}
+
+/**************************************************
+          BUILD STEP
+ **************************************************/
+async function build() {
+  log(`🚀 Building ${edition} jar for ${version} from commit ${commit}`);
+
+  // check build environment
+  const majorVersion = Number(getMajorVersion(version));
+  const nodeVersion = (await $`node --version`).toString();
+  const javaVersion = (await $`java -version`).toString();
+
+  if (!nodeVersion.includes(`v${versionRequirements[majorVersion].node}`)) {
+    error(
+      `Node version must be v${versionRequirements[majorVersion].node}, you are running ${nodeVersion}`,
+    );
+  }
+
+  if (
+    !javaVersion.includes(
+      `openjdk version "${versionRequirements[majorVersion].java}`,
+    )
+  ) {
+    error(
+      `Java version must be ${versionRequirements[majorVersion].java}, you are running ${javaVersion}`,
+    );
+  }
+
+  const currentBranch = (await $`git branch --show-current`).toString().trim();
+
+  try {
+    await $`git fetch --all`;
+    await $`git stash && git checkout ${commit}`;
+
+    // actually build jar
+    await $`../bin/build.sh :edition :${edition} :version ${version}`.pipe(
+      process.stdout,
+    );
+
+    await $`git checkout -`;
+    await $`echo ${commit} > ${JAR_PATH}/COMMIT-ID`;
+    await $`shasum -a 256 ${JAR_PATH}/metabase.jar > ${JAR_PATH}/SHA256.sum`;
+
+    log(`✅ Built ${edition} jar for ${version} in ${JAR_PATH}`, "green");
+  } catch (error) {
+    console.error(error);
+  } finally {
+    await $`git checkout ${currentBranch}`;
+  }
+}
+
+/**************************************************
+          PUBLISH STEPS
+***************************************************/
+
+async function s3() {
+  log(`⏳ Publishing ${version} to s3`);
+
+  const { GITHUB_OWNER, GITHUB_REPO } = getGithubCredentials();
+
+  await checkJar();
+
+  const isLatest = await isLatestRelease({
+    github,
+    owner: GITHUB_OWNER,
+    repo: GITHUB_REPO,
+    version,
+  });
+
+  if (
+    !AWS_S3_DOWNLOADS_BUCKET ||
+    !AWS_S3_STATIC_BUCKET ||
+    !AWS_S3_RELEASE_SECRET_ACCESS_KEY ||
+    !AWS_S3_RELEASE_ACCESS_KEY_ID
+  ) {
+    error("You must provide all AWS environment variables in .env-template");
+  }
+
+  process.env.AWS_ACCESS_KEY_ID = AWS_S3_RELEASE_ACCESS_KEY_ID;
+  process.env.AWS_SECRET_ACCESS_KEY = AWS_S3_RELEASE_SECRET_ACCESS_KEY;
```

Comment:
What's the purpose of aliasing those?

Can we just have `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` in `.env-template`?